service/worker/batcher/activities.go

```diff
@@ -464,6 +601,248 @@ func startTaskProcessor(
+							Namespace: batchParams.Namespace,
+							Execution: &commonpb.WorkflowExecution{
+								WorkflowId: workflowID,
+								RunId:      runID,
+							},
+							Identity:               operation.ResetActivitiesOperation.Identity,
+							ResetHeartbeat:         operation.ResetActivitiesOperation.ResetHeartbeat,
+							Jitter:                 operation.ResetActivitiesOperation.Jitter,
+							KeepPaused:             operation.ResetActivitiesOperation.KeepPaused,
+							RestoreOriginalOptions: operation.ResetActivitiesOperation.RestoreOriginalOptions,
+						}
+
+						switch ao := operation.ResetActivitiesOperation.GetActivity().(type) {
+						case *batchpb.BatchOperationResetActivities_Type:
+							resetRequest.Activity = &workflowservice.ResetActivityRequest_Type{Type: ao.Type}
+						case *batchpb.BatchOperationResetActivities_MatchAll:
+							resetRequest.Activity = &workflowservice.ResetActivityRequest_MatchAll{MatchAll: true}
+						}
+
+						_, err = frontendClient.ResetActivity(ctx, resetRequest)
+						return err
+					})
+			case *batchpb.BatchOperation_UpdateActivityOptionsOperation:
+				err = processTask(ctx, limiter, task,
+					func(workflowID, runID string) error {
+						updateRequest := &workflowservice.UpdateActivityOptionsRequest{
+							Namespace: batchParams.Namespace,
+							Execution: &commonpb.WorkflowExecution{
+								WorkflowId: workflowID,
+								RunId:      runID,
+							},
+							UpdateMask:      &fieldmaskpb.FieldMask{Paths: operation.UpdateActivityOptionsOperation.UpdateMask.Paths},
+							RestoreOriginal: operation.UpdateActivityOptionsOperation.RestoreOriginal,
+							Identity:        operation.UpdateActivityOptionsOperation.Identity,
+						}
+
+						switch ao := operation.UpdateActivityOptionsOperation.GetActivity().(type) {
+						case *batchpb.BatchOperationUpdateActivityOptions_Type:
+							updateRequest.Activity = &workflowservice.UpdateActivityOptionsRequest_Type{Type: ao.Type}
+						case *batchpb.BatchOperationUpdateActivityOptions_MatchAll:
+							updateRequest.Activity = &workflowservice.UpdateActivityOptionsRequest_MatchAll{MatchAll: true}
+						}
+
+						if ao := operation.UpdateActivityOptionsOperation.GetActivityOptions(); ao != nil {
+							updateRequest.ActivityOptions = &activitypb.ActivityOptions{
+								ScheduleToStartTimeout: ao.ScheduleToStartTimeout,
+								ScheduleToCloseTimeout: ao.ScheduleToCloseTimeout,
+								StartToCloseTimeout:    ao.StartToCloseTimeout,
+								HeartbeatTimeout:       ao.HeartbeatTimeout,
+							}
+
+							if rp := ao.RetryPolicy; rp != nil {
+								updateRequest.ActivityOptions.RetryPolicy = &commonpb.RetryPolicy{
+									InitialInterval:        rp.InitialInterval,
+									BackoffCoefficient:     rp.BackoffCoefficient,
+									MaximumInterval:        rp.MaximumInterval,
+									MaximumAttempts:        rp.MaximumAttempts,
+									NonRetryableErrorTypes: rp.NonRetryableErrorTypes,
+								}
+							}
+						}
```

Comment:
Better just pass through and not have to remember to propagate new options as they're added

```suggestion
						updateRequest.ActivityOptions = operation.UpdateActivityOptionsOperation.GetActivityOptions()
```